docs(expo-widgets): create config plugin documentation (#42311)
expo/expo: docs/pages/versions/unversioned/sdk/widgets.mdx

---
title: Widgets

> description: A config plugin that enables iOS home screen widgets and Live Activities in Expo apps.

Reviewer: Not sure how to phrase it, but it's more than just a config plugin.
```suggestion
description: A library that provides iOS home screen widgets and Live Activities in Expo apps.
```

    {
      name: 'widgets[].description',
      description:
        "A brief description of what the widget does. This appears in the widget gallery to help users understand the widget's purpose.",
    },
    {
      name: 'widgets[].supportedFamilies',
      description: [
        'An array of widget sizes that this widget supports. Available options:',
        '* `systemSmall` - Small square widget (2x2 grid)',
        '* `systemMedium` - Medium rectangular widget (4x2 grid)',
        '* `systemLarge` - Large square widget (4x4 grid)',
        '* `systemExtraLarge` - Extra large widget (iPad only, 6x4 grid)',
        '* `accessoryCircular` - Circular widget for Lock Screen and Apple Watch',
        '* `accessoryRectangular` - Rectangular widget for Lock Screen and Apple Watch',
        '* `accessoryInline` - Inline text widget for Lock Screen complications',
      ].join('\n'),
    },
  ]}
/>

### Full example with all options

```json app.json
{
  "expo": {
    "plugins": [
      [
        "expo-widgets",
        {
          "bundleIdentifier": "com.example.myapp.widgets",
          "groupIdentifier": "group.com.example.myapp",
          "enablePushNotifications": true,

>           "frequentUpdates": true,

Reviewer: ```suggestion
```

---
title: Widgets
description: A config plugin that enables iOS home screen widgets and Live Activities in Expo apps.
sourceCodeUrl: 'https://github.com/expo/expo/tree/main/packages/expo-widgets'
packageName: 'expo-widgets'
platforms: ['ios']
---

import { APIInstallSection } from '~/components/plugins/InstallSection';
import { ConfigPluginExample, ConfigPluginProperties } from '~/ui/components/ConfigSection';

> `expo-widgets` is a [config plugin](/config-plugins/introduction/) that enables iOS home screen widgets and Live Activities in your Expo app. It configures the necessary native project settings during [Prebuild](/workflow/prebuild), including creating the widget extension target, setting up app groups for data sharing between your app and widgets, and configuring entitlements.

Reviewer 2: ```suggestion
`expo-widgets` is a [config plugin](/config-plugins/introduction/) that enables iOS home screen widgets and Live Activities in your Expo app. It configures the necessary native project settings during [Prebuild](/workflow/continuous-native-generation/#prebuild), including creating the widget extension target, setting up app groups for data sharing between your app and widgets, and configuring entitlements.
```

That link is just a redirect. Let's replace it with the correct link.

        },
        {
          name: 'displayName',
          description:
            'The user-facing name of the widget that appears in the widget gallery when users add widgets to their home screen.',
        },
        {
          name: 'description',
          description:
            "A brief description of what the widget does. This appears in the widget gallery to help users understand the widget's purpose.",
        },
        {
          name: 'supportedFamilies',
          description: [
            'An array of widget sizes that this widget supports. Available options:',
            '* `systemSmall` - Small square widget (2x2 grid)',
            '* `systemMedium` - Medium rectangular widget (4x2 grid)',
            '* `systemLarge` - Large square widget (4x4 grid)',
            '* `systemExtraLarge` - Extra large widget (iPad only, 6x4 grid)',
            '* `accessoryCircular` - Circular widget for Lock Screen and Apple Watch',
            '* `accessoryRectangular` - Rectangular widget for Lock Screen and Apple Watch',
            '* `accessoryInline` - Inline text widget for Lock Screen complications',
          ].join('\n'),
        },
      ],

>     },

Reviewer 2: ```suggestion
    {
      name: 'widgets',
      description:
        'An array of widget configurations. Each widget in the array will be generated as a separate widget kind in your widget extension.',
    },
    {
      name: 'widgets[].name',
      description:
        'The internal name (identifier) of the widget. This is used as the Swift struct name and should be a valid Swift identifier (no spaces or special characters).',
    },
    {
      name: 'widgets[].displayName',
      description:
        'The user-facing name of the widget that appears in the widget gallery when users add widgets to their home screen.',
    },
    {
      name: 'widgets[].description',
      description:
        "A brief description of what the widget does. This appears in the widget gallery to help users understand the widget's purpose.",
    },
    {
      name: 'widgets[].supportedFamilies',
      description: [
        'An array of widget sizes that this widget supports. Available options:',
        '* `systemSmall` - Small square widget (2x2 grid)',
        '* `systemMedium` - Medium rectangular widget (4x2 grid)',
        '* `systemLarge` - Large square widget (4x4 grid)',
        '* `systemExtraLarge` - Extra large widget (iPad only, 6x4 grid)',
        '* `accessoryCircular` - Circular widget for Lock Screen and Apple Watch',
        '* `accessoryRectangular` - Rectangular widget for Lock Screen and Apple Watch',
        '* `accessoryInline` - Inline text widget for Lock Screen complications',
      ].join('\n'),
    },
```

Sub array of properties in `ConfigPluginProperties` is not supported, so this section will not display. Let's update it to the above so it renders correctly:

<img width="2330" height="2250" alt="CleanShot 2026-01-22 at 03 47 15@2x" src="https://github.com/user-attachments/assets/2f839eb7-750d-42b9-8670-af8750d88dd3" />
